enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -761,6 +763,17 @@ func ConvertFileToCOW(ctx context.Context, env environment.Env, filePath string,
 	return NewCOWStore(ctx, env, chunks, chunkSizeBytes, totalSizeBytes, dataDir, remoteInstanceName, remoteEnabled)
 }
 
+// getMmapLRU returns the shared LRU instance to be used for the mmap,
+// if applicable.
+func getMmapLRU(dataDir string) (*MmapLRU, error) {
+	// We constrain UFFD memory usage by only allowing one chunk to be mapped
+	// at once, so don't use the shared LRU for UFFD.
+	if filepath.Base(dataDir) != snaputil.MemoryFileName {
```

Comment:
This code is probably the grossest thing about this PR (depending in the file name to determine whether we use the shared LRU) but there are two reasons why I think it's OK:

* Avoids extra plumbing in snaploader / NewCOWStore / NewMmap* and keeps the PR smol
* In today's world, snaploader would just have to use the chunk file name anyway because that's how we identify the memory snapshot vs. the disk snapshot.